contracts/test/Token.t.sol

```diff
@@ -0,0 +1,348 @@
+        Token token = new Token(name, symbol, decimals);
+
+        assertEq(token.name(), name);
+        assertEq(token.symbol(), symbol);
+        assertEq(token.decimals(), decimals);
+    }
+
+    function testFuzz_mint(address account, uint256 amount) public {
+        vm.assume(account != address(0));
+
+        Token token = new Token("", "", 18);
+
+        token.mint(account, amount);
+
+        assertEq(token.totalSupply(), amount);
+        assertEq(token.balanceOf(account), amount);
+    }
+
+    function testFuzz_burn(address account, uint256 balance, uint256 burnAmount) public {
+        vm.assume(account != address(0));
+        vm.assume(balance >= burnAmount);
+
+        Token token = new Token("", "", 18);
+
+        token.mint(account, balance);
+        token.burn(account, burnAmount);
+
+        assertEq(token.totalSupply(), balance - burnAmount);
+        assertEq(token.balanceOf(account), balance - burnAmount);
+    }
+
+    function testFuzz_approve(address account, uint256 amount) public {
+        vm.assume(account != address(0));
+
+        Token token = new Token("", "", 18);
+
+        assertTrue(token.approve(account, amount));
+        assertEq(token.allowance(address(this), account), amount);
+    }
+
+    function testFuzz_transferCorrectlyUpdatesBalances(
+        address sender,
+        address receiver,
+        uint256 mintAmount,
+        uint256 transferAmount
+    ) public {
+        vm.assume(sender != address(0) && receiver != address(0) && sender != receiver);
+        vm.assume(mintAmount > 0 && transferAmount > 0 && mintAmount >= transferAmount);
+
+        Token token = new Token("", "", 18);
+
+        token.mint(sender, mintAmount);
+
+        uint256 initialSenderBalance = token.balanceOf(sender);
+        uint256 initialReceiverBalance = token.balanceOf(receiver);
+        uint256 initialTotalSupply = token.totalSupply();
+
+        vm.prank(sender);
+        token.transfer(receiver, transferAmount);
+
+        assertEq(token.balanceOf(sender), initialSenderBalance - transferAmount);
+        assertEq(token.balanceOf(receiver), initialReceiverBalance + transferAmount);
+        assertEq(token.totalSupply(), initialTotalSupply);
+    }
+
+    function testFuzz_transferFrom(
+        address owner,
+        address spender,
+        address receiver,
+        uint256 mintAmount,
+        uint256 allowanceAmount,
+        uint256 transferAmount
+    ) public {
+        // Avoid zero address and ensure distinct addresses
+        vm.assume(owner != address(0) && spender != address(0) && receiver != address(0));
+        vm.assume(owner != spender && spender != receiver && owner != receiver);
+
+        // Ensure amounts are valid
+        vm.assume(mintAmount > 0 && allowanceAmount > 0 && transferAmount > 0);
+        vm.assume(mintAmount >= transferAmount);
+        vm.assume(allowanceAmount >= transferAmount);
```

Comment:
We need a simple test that rejects if the allowance is less than the amount transferred. The fuzz tests assume that the amount is always greater than allowance so we are not covered there.

Reply:
Added in https://github.com/Snowfork/snowbridge/pull/1384/commits/ded3f51d0d854b7b45eb98d984beccc7a7f3c384